src/intersections/polygons.jl

```diff
@@ -3,14 +3,7 @@
 # ------------------------------------------------------------------
 
 function intersection(f, poly₁::Polygon, poly₂::Polygon)
-  # TODO: use Weiler-Atherton or other more general clipping method
-  clipped = if isconvex(poly₂)
-    clip(poly₁, poly₂, SutherlandHodgmanClipping())
-  elseif isconvex(poly₁)
-    clip(poly₂, poly₁, SutherlandHodgmanClipping())
-  else
-    throw(ErrorException("intersection not implemented between two non-convex polygons"))
-  end
```

Comment:
In order to control risk of new bugs, we need to be extra careful with modifications that have wide impact in the project. Our contribution guidelines explain that a PR should do one thing and one thing only to facilitate review and reduce side effects.

Please remove this change in the intersection function, and focus on the new `clip` method with tests.